esp/platform/txsummary.hpp

```diff
@@ -50,80 +75,264 @@ class CTxSummary : extends CInterface
+
+        // Some Entry subclasses can assign a suffix to serialize
+        // after its value
+        StringAttr suffix;
+
+        // The minimum logLevel at which this entry is serialized
         LogLevel logLevel = LogMax;
 
-        Entry(const char* _key, const char* _value, const LogLevel _logLevel) : key(_key), value(_value), logLevel(_logLevel) {}
+        // The bit field indicating with groups this entry belongs to
+        unsigned int group = DEFAULT_GROUP;
+
+        TxEntryBase(const char* _key, const LogLevel _logLevel, const unsigned int _group = DEFAULT_GROUP, const char* _suffix = nullptr);
+        virtual bool contains(const char* path) {return false;}
+        virtual TxEntryBase* queryEntry(const char* path) {return nullptr;}
+        virtual TxEntryBase* ensurePath(const char* path) {return nullptr;}
+        virtual bool append(TxEntryBase* entry) {return false;}
+        virtual bool set(TxEntryBase* entry) {return false;}
+        virtual unsigned __int64 size() {return 1;}
+        virtual void clear() {};
+        virtual bool shouldSerialize(const LogLevel requestedLevel, const unsigned int requestedGroup);
+        virtual StringBuffer& serialize(StringBuffer& buf, const LogLevel requestedLevel, const unsigned int requestedGroup, const unsigned int requestedStyle);
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr){return nullptr;};
+    };
+
+    struct TxEntryStr : public TxEntryBase
+    {
+        StringAttr value;
+
+        TxEntryStr(const char* _key, const char* _value, const LogLevel _logLevel, const unsigned int _group = DEFAULT_GROUP)
+            : TxEntryBase(_key, _logLevel, _group), value(_value) {}
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr) override {return new TxEntryStr(newName ? newName : fullname.get(), value.get(), logLevel, group);}
+
+        // When the string value is nullptr, it is serialized to JSON as the JSON null value
+        virtual StringBuffer& serialize(StringBuffer& buf, const LogLevel requestedLevel, const unsigned int requestedGroup, const unsigned int requestedStyle) override;
     };
 
-    using Entries = std::list<Entry>;
-    using TimerKey = StringAttr;
-    using TimerValue = Linked<CumulativeTimer>;
-    using Timers = std::map<TimerKey, TimerValue>;
+    struct TxEntryBool : public TxEntryBase
+    {
+        bool value;
+
+        TxEntryBool(const char* _key, bool _value, const LogLevel _logLevel, const unsigned int _group = DEFAULT_GROUP)
+            : TxEntryBase(_key, _logLevel, _group), value(_value) {}
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr) override {return new TxEntryBool(newName ? newName : fullname.get(), value, logLevel, group);}
+        virtual StringBuffer& serialize(StringBuffer& buf, const LogLevel requestedLevel, const unsigned int requestedGroup, const unsigned int requestedStyle) override;
+    };
+
+    struct TxEntryNumber : public TxEntryBase
```

Comment:
This could be made a template, with the template parameter being the numeric type. No subclasses necessary.

Reply:
removed numeric subclasses